### PR TITLE
abacus：add new package

### PR DESCRIPTION
**Pull request by Bitllion**
Add abacus installation script and modify some compilation parameters of elpa package in MPI for it

### CODE REVIEW

**Comment by Bitllion** [comment]
@tldahlgren @alalazo

File: var/spack/repos/builtin/packages/abacus/package.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack import *
+import os
+class Abacus(MakefilePackage):
```
Comment (lhxone) [inline var/spack/repos/builtin/packages/abacus/package.py:9]: 这一行之前应该有两个空行

File: var/spack/repos/builtin/packages/abacus/package.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    """ABACUS (Atomic-orbital Based Ab-initio Computation at UStc) is an open-source computer code package aiming for large-scale electronic-structure simulations from first principles"""
```
Comment (lhxone) [inline var/spack/repos/builtin/packages/abacus/package.py]: 每一行的字不能太多，行尾不可以有多余的空格，简介可以分成多行来写

File: var/spack/repos/builtin/packages/abacus/package.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    conflicts('+openmp',
+              when='^intel-oneapi-mkl',
+              msg=mkl_message)
+
+    build_directory = 'source'
+
+    def edit(self, spec, prefix):
+        if self.version == Version('2.2.1'):
+            makefile_vars = FileFilter('source/Makefile.vars')
+            makefile_vars.filter('^\s*LAPACK_DIR\s*=.*',  'LAPACK_DIR = '+ spec['mkl'].prefix)
```
Comment (lhxone) [inline var/spack/repos/builtin/packages/abacus/package.py]: 可以使用[autopep8](https://github.com/hhatto/autopep8)来格式化代码

File: var/spack/repos/builtin/packages/elpa/package.py [hunk 4]
```diff
@@ -179,4 +181,4 @@ def configure_args(self):
         options.append('--disable-silent-rules')
         options.append('--without-threading-support-check-during-build')
 
-        return options
+        return options
```
Comment (lhxone) [inline var/spack/repos/builtin/packages/elpa/package.py]: 在文件最后要有一行空白行

**Review by lhxone — COMMENTED** [review]
Style check not passed,here is some advice.

File: var/spack/repos/builtin/packages/elpa/package.py [hunk 5]
```diff
@@ -167,9 +168,10 @@ def configure_args(self):
 
         if '+mpi' in self.spec:
             options += [
-                'CC={0}'.format(spec['mpi'].mpicc),
-                'CXX={0}'.format(spec['mpi'].mpicxx),
-                'FC={0}'.format(spec['mpi'].mpifc),
+                'CC=mpiicc',
+                'CXX=mpiicpc',
+                'FC=mpiifort',
+                'FCFLAGS=-mkl=cluster',
```
Comment (dev-zero) [inline var/spack/repos/builtin/packages/elpa/package.py:173]: NACK

File: var/spack/repos/builtin/packages/elpa/package.py [hunk 6]
```diff
@@ -11,6 +11,7 @@
 class Elpa(AutotoolsPackage, CudaPackage, ROCmPackage):
     """Eigenvalue solvers for Petaflop-Applications (ELPA)"""
 
+
```
Comment (dev-zero) [inline var/spack/repos/builtin/packages/elpa/package.py]:
```suggestion
```

**Review by dev-zero — CHANGES_REQUESTED** [review]
This makes the elpa build Intel-dependent? So, NACK.

File: var/spack/repos/builtin/packages/elpa/package.py (hunk 5 above)
Comment (Bitllion) [reply var/spack/repos/builtin/packages/elpa/package.py:173]: Yes, I am the person in charge of compiling and installing this software on the HPC cluster. For the best effect, I only test the Intel compiler

**Pull request closed without merging**